tests/upgrade/conftest.py

```
@@ -0,0 +1,50 @@
+        raise ValueError("upgrade_repo_ref must be provided via --tc=upgrade_repo_ref:<ref>")
+    if not script_relative_path:
+        raise ValueError("upgrade_script_path must be provided via --tc=upgrade_script_path:<relative-path>")
+
+    clone_dir = tmp_path_factory.mktemp("mtv-autodeploy")
+
+    LOGGER.info(f"Cloning {repo_url} (ref={repo_ref}) into {clone_dir}")
+    Repo.clone_from(url=repo_url, to_path=str(clone_dir), branch=repo_ref, depth=1)
+
+    script_path = clone_dir / script_relative_path
+    if not script_path.is_file():
+        raise ValueError(f"Upgrade script not found at '{script_path}' after cloning {repo_url}")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)